src/toxcore/onion/packet/onion_data_response.rs

```diff
@@ -66,7 +70,7 @@ impl OnionDataResponse {
     /// Create `OnionDataResponse` from `OnionDataResponsePayload` encrypting it
     /// with `shared_key` and `nonce`
     pub fn new(shared_secret: &PrecomputedKey, temporary_pk: PublicKey, nonce: Nonce, payload: &OnionDataResponsePayload) -> OnionDataResponse {
-        let mut buf = [0; 293];
```

Comment:
293 - 245 = 48

MAX_ONION_RESPONSE_PAYLOAD_SIZE = 293 (correct?)
MAX_ONION_RESPONSE_PAYLOAD_SIZE = MAX_ONION_CLIENT_DATA_SIZE + PUBLICKEYBYTES

MAX_ONION_CLIENT_DATA_SIZE = 245 (correct?)

so PUBLICKEYBYTES = MAX_ONION_RESPONSE_PAYLOAD_SIZE - MAX_ONION_CLIENT_DATA_SIZE = 48,

but 

```
#define crypto_box_curve25519xsalsa20poly1305_PUBLICKEYBYTES 32U
```

Where is the truth?

Reply:
293 and 245 are for DhtPkAnnounce only.
These numbers is correct if we use only DhtPkAnnounce.
Now, we added FriendRequest, so this array must be entended.